src/logion/model/supportedaccountid.model.ts

```diff
@@ -1,52 +1,82 @@
 import { components } from "../controllers/components.js";
 import { Column } from "typeorm";
+import { ValidAccountId, AnyAccountId, AccountType, AccountId } from "@logion/node-api";
 
 export type SupportedAccountId = components["schemas"]["SupportedAccountId"];
-type AddressType = components["schemas"]["AddressType"];
 
-export class EmbeddableSupportedAccountId {
+export class EmbeddableNullableAccountId {
 
-    @Column({ length: 255, name: "_address_type" })
+    @Column({ length: 255, name: "_address_type", nullable: true })
     type?: string;
 
-    @Column({ length: 255, name: "_address" })
+    @Column({ length: 255, name: "_address", nullable: true })
     address?: string;
 
-    toSupportedAccountId(): SupportedAccountId {
-        return {
-            type: (this.type || "Polkadot") as AddressType,
-            address: this.address || "",
+    toValidAccountId(): ValidAccountId | undefined {
+        if (this.type && this.address) {
+            const accountId = new AnyAccountId(
+                this.address,
+                this.type as AccountType
+            );
+            return accountId.toValidAccountId();
+        } else {
+            return undefined;
         }
     }
 
-    static from(supportedAccountId: SupportedAccountId): EmbeddableSupportedAccountId {
-        const embeddable = new EmbeddableSupportedAccountId();
-        embeddable.type = supportedAccountId.type;
-        embeddable.address = supportedAccountId.address;
+    static from(validAccountId: ValidAccountId): EmbeddableNullableAccountId {
+        const embeddable = new EmbeddableNullableAccountId();
+        embeddable.type = validAccountId.type;
+        embeddable.address = validAccountId.getAddress(DB_SS58_PREFIX);
         return embeddable;
     }
 }
 
-type AnyAccountId = {
-    type?: string | AddressType;
+export class EmbeddableAccountId {
+
+    @Column({ length: 255, name: "_address_type" })
+    type?: string;
+
+    @Column({ length: 255, name: "_address" })
     address?: string;
+
+    toValidAccountId(): ValidAccountId {
+        const accountId = new AnyAccountId(
+            this.address || "",
+            (this.type || "Polkadot") as AccountType
+        );
+        return accountId.toValidAccountId();
+    }
+
+    static from(validAccountId: ValidAccountId): EmbeddableAccountId {
+        const embeddable = new EmbeddableAccountId();
+        embeddable.type = validAccountId.type;
+        embeddable.address = validAccountId.getAddress(DB_SS58_PREFIX);
+        return embeddable;
+    }
 }
 
-export function accountEquals(left: AnyAccountId | undefined, right: AnyAccountId | undefined): boolean {
-    if (
-        left === undefined || right === undefined ||
-        left.address === undefined || right.address === undefined ||
-        left.type === undefined || right.type === undefined
-    ) {
+export function accountEquals(left: ValidAccountId | undefined, right: ValidAccountId | undefined): boolean {
```

Comment:
We probably can get rid of this function and use `ValidAccountId.equals(...)` instead.

Reply:
Indeed, done.